modules/standard/Map.chpl

```diff
@@ -207,7 +207,8 @@ module Map {
     }
 
     pragma "no doc"
-    proc const this(k: keyType) const {
+    proc const this(k: keyType) const
+    where shouldReturnRvalueByValue(valType) {
```

Comment:
This is fine for now but in general I was expecting that shouldReturnRvalueByValue would only be used with arrays. (I personally think we should move away from it, and always return by `const ref`).